crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -106,10 +116,24 @@ impl CumulocityConverter {
 
     fn try_convert_event(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> {
         let tedge_event = ThinEdgeEvent::try_from(input.topic.name.as_str(), input.payload_str()?)?;
-        let smartrest_alarm = event::serialize_event(tedge_event)?;
-        let smartrest_topic = Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC);
+        dbg!(input.payload_bytes().len());
+        if input.payload_bytes().len() > MQTT_EVENT_SIZE_LIMIT {
+            let event_text = tedge_event
+                .data
+                .and_then(|data| data.message)
+                .unwrap_or_else(|| "generic event".into());
+            let _ = block_on(self.http_proxy.send_event(
+                tedge_event.name.as_str(),
+                event_text.as_str(),
+                None,
+            ))?;
+            Ok(vec![])
```

Comment:
To be honest I'm not a fan of this pattern that we do something before and than send a result with empty vec.

I would be more happy if the `http proxy` would be running as another task and this task would only dispatch a message over a channel with required events.

Reply:
Yeah, that's a good suggestion. I was not a big fan of injecting the http proxy into the converters either, which I had expressed in the [mapper merge PR](https://github.com/thin-edge/thin-edge.io/pull/789#discussion_r802338632) as well. I'd like to have a similar abstraction that we have with MQTT for HTTP messaging as well, where the converters just produce the messages and and http message processor, running as a separate task, picks these and sends them one-by-one.

But, I've got many concerns around that design, especially around the inherent request-response model of HTTP. While using HTTP in our mappers, one message would often be the result of/derived from a previous message's response (like in the log request case where we first send a log event message and using the event id in its response send another message attaching the log binary). So, the interaction between the converter and the http message processor can't be just a one-way channel, like we have for MQTT messages. It'll require some involved design thinking and hence I'd prefer to address that in a separate standalone issue/PR.

Reply:
I created a specific discussion for this idea of an [HTTP proxy for the cloud end-point REST API](https://github.com/thin-edge/thin-edge.io/discussions/915).